test/test_python_dispatch.py

```diff
@@ -352,6 +363,179 @@ def sqsum(a: SymInt, b: SymInt):
+            'a::b(Tensor x) -> (Tensor y, Tensor z)',
+            'a::b(Tensor x, Tensor w) -> (Tensor y, Tensor z)',
+            'a::b.with_overload(Tensor x, Tensor w) -> (Tensor, Tensor)',
+            'a::b(Tensor x, Tensor w) -> Tensor',
+            'a::b(Tensor? x, Tensor w) -> Tensor',
+            'a::b(Tensor? x, Tensor[] w) -> Tensor',
+            'a::b(Tensor x, int[] w) -> Tensor',
+            'a::b(Tensor x, SymInt[] w) -> Tensor',
+            'a::b(Tensor x, Scalar w) -> Tensor',
+            'a::b(Tensor x, float w) -> Tensor',
+            'a::b(Tensor x, float? w) -> Tensor',
+            'a::b(Tensor x, bool[] w) -> Tensor',
+        ]
+
+        for schema in schemas:
+            CustomOp.define(schema)
+
+    def test_reserved_ns(self):
+        from torch._custom_op import RESERVED_NS
+
+        for ns in RESERVED_NS:
+            with self.assertRaisesRegex(ValueError, 'is a reserved namespace'):
+                CustomOp.define(f'{ns}::a(Tensor x) -> Tensor')
+
+    def test_private_ctor(self):
+        with self.assertRaisesRegex(RuntimeError, 'CustomOp constructor is private'):
+            CustomOp(None, None, None)
+
+    def test_lifetime_is_tied_to_object(self):
+        foo = CustomOp.define('custom::foo(Tensor x) -> Tensor')
+
+        # If there is more than one reference then we've got a problem - something
+        # will keep the object alive.
+        # NB: sys.getrefcount(foo) reports one more than the number (since it creates a ref)
+        self.assertEqual(sys.getrefcount(foo), 2)
+
+        # We can't define an op multiple times,
+        with self.assertRaisesRegex(RuntimeError, 'multiple times'):
+            bar = CustomOp.define('custom::foo(Tensor x) -> Tensor')
+
+        # Unless we delete the original op.
+        del foo
+        baz = CustomOp.define('custom::foo(Tensor x) -> Tensor')
+
+    def test_autograd_notimplemented(self):
+        foo = CustomOp.define('custom::foo(Tensor x) -> Tensor')
+        x = torch.randn(3, requires_grad=True)
+        with self.assertRaisesRegex(RuntimeError, 'Autograd has not been implemented'):
+            foo(x)
+        del foo
+
+        foo = CustomOp.define('custom::foo(Tensor[] xs) -> Tensor')
+        x = torch.randn(3, requires_grad=True)
+        y = torch.randn(3)
+        with self.assertRaisesRegex(RuntimeError, 'Autograd has not been implemented'):
+            foo([y, x])
+        del foo
+
+        foo = CustomOp.define('custom::foo(Tensor x, Tensor y) -> Tensor')
+        x = torch.randn(3, requires_grad=True)
+        y = torch.randn(3)
+        with self.assertRaisesRegex(RuntimeError, 'Autograd has not been implemented'):
+            foo(y, x)
+        del foo
+
+    def test_impl_cpu(self):
+        foo = CustomOp.define('custom::foo.blahblahblah(Tensor x) -> Tensor')
+
+        @foo.impl('cpu')
+        def foo_cpu(x):
+            return x.sin()
+
+        x = torch.randn(3)
+        result = foo(x)
```

Comment:
Just wondering, can we directly invoke the overload? Previously we could do like:
```
torch.ops.custom_lib.foo
```

Reply:
Invoking torch.ops.custom_lib.foo is still supported, though I'm not sure why one would want to grab it directly and invoke it. Do you see use cases for this?